### PR TITLE
[CELEBORN-1711][TEST] Fix '65535' port is invalid

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?

Fix flaky test caused by invalid port. 
```
[info] ApiMasterResourceSuite:
[info] org.apache.celeborn.service.deploy.master.http.api.ApiMasterResourceSuite *** ABORTED ***
[info]   java.lang.IllegalArgumentException: '65535' in celeborn.master.http.port is invalid. Invalid port
```

### Why are the changes needed?

The ports range in CelebornConf is [1024, 65535), 65535 is excluded.

https://github.com/apache/celeborn/blob/169b6f6973b2ee5093d91df0d2b573977efdc7ae/common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala#L2315-L2324

image

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?

GA.

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]
the test change looks good.

could you please also update the method definition to clarify the inclusive and exclusive of each parameter?

```
def selectRandomPort(from: Int, to: Int): Int
```

File: common/src/main/scala/org/apache/celeborn/common/util/Utils.scala [hunk 1]
```diff
@@ -245,8 +245,15 @@ object Utils extends Logging {
...
-    ScalaRandom.nextInt(to - from) + from
+  /**
+   * Selects a random port within the specified range.
+   *
+   * @param from the lower bound of the range (inclusive)
+   * @param until the upper bound of the range (exclusive)
+   * @return a randomly selected port within the range [from, until)
+   */
+  def selectRandomPort(from: Int, until: Int): Int = {
+    ScalaRandom.nextInt(until - 1 - from) + from
```
Comment (turboFei) [inline common/src/main/scala/org/apache/celeborn/common/util/Utils.scala:256]: image

Due this method is only used by UT, both for scala and java UTs.

To minimize the code change, I just change the `selectRandomPort`  method to make the upper bound be exclusive.
cc @pan3793

File: common/src/main/scala/org/apache/celeborn/common/util/Utils.scala [hunk 2]
```diff
@@ -245,8 +245,15 @@ object Utils extends Logging {
...
-  def selectRandomPort(from: Int, to: Int): Int = {
-    ScalaRandom.nextInt(to - from) + from
+  /**
+   * Selects a random port within the specified range.
+   *
+   * @param from the lower bound of the range (inclusive)
+   * @param until the upper bound of the range (exclusive)
+   * @return a randomly selected port within the range [from, until)
+   */
+  def selectRandomPort(from: Int, until: Int): Int = {
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/util/Utils.scala:255]: maybe `randomInt`? it does not have specific logic for "port"

**Review by pan3793 — APPROVED** [review]

**Review by SteNicholas — APPROVED** [review]

**Review by SteNicholas — APPROVED** [review]

**Comment by turboFei** [comment]
The invalid port issue happen frequently recently. 
I plan to merge it and then rebase https://github.com/apache/celeborn/pull/2906 on this, thanks all for the review.

**Pull request closed without merging**